contracts/cw721-piggy-bank/src/contract.rs

```diff
@@ -32,11 +34,14 @@ pub fn instantiate(
 ) -> StdResult<Response> {
     cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
-    // TODO Validate denoms are formated correctly
-
+    // Validate denoms are formatted correctly
+    let unchecked_denom = UncheckedDenom::Native(msg.deposit_denom.clone());
+    let _checked_denom = unchecked_denom.into_checked(deps.as_ref()).map_err(|_| StdError::generic_err("Invalid deposit denom"))?;
```

Comment:
In rust, we typically use the `_` prefix for a variable if it's not being used.

Consider:
```suggestion
    let checked_denom = unchecked_denom.into_checked(deps.as_ref()).map_err(|_| StdError::generic_err("Invalid deposit denom"))?;
```